sympy/polys/rootoftools.py

```diff
@@ -759,6 +762,32 @@ def _get_roots(cls, method, poly, radicals):
             roots.append(coeff*cls._postprocess_root(root, radicals))
         return roots
 
+    @classmethod
+    def _roots_Algebraic(cls, poly):
+        r"""Converts a Polynomial 'B(x)' with algebriac coefficients
+        to a Polynomial 'C(x)' with rational coefficients such that
+        every root of 'B(x)' is a root of 'C(x)'.
+        """
+        from sympy.polys.numberfields import primitive_element
+        from sympy.polys.polytools import gcd, resultant, reduced, div
+        x, y = symbols('x y')
+        coeffs = Poly(poly, x).all_coeffs()
+        try:
+            A, r_A, B_s = primitive_element(coeffs, y, ex=True)
```

Comment:
Better take only those coeffs that are not rational to speed up primitive_element.

Reply:
Yeah, maybe that can be done.